src/main/java/org/checkerframework/checker/startswith/StartsWithAnnotatedTypeFactory.java

```diff
@@ -180,13 +175,27 @@ public boolean isSubtype(AnnotationMirror subAnno, AnnotationMirror superAnno) {
                     && AnnotationUtils.hasElementValue(superAnno, "value")) {
                 List<String> subArrays = AnnotationUtils.getElementValueArray(subAnno, "value", String.class, true);
                 List<String> superArrays = AnnotationUtils.getElementValueArray(superAnno, "value", String.class, true);;
-                if (superArrays.containsAll(subArrays)) {
-                    return true;
-                }
+                return checkStartsWith(subArrays, superArrays);
             }
             return false;
         }
 
+        private boolean checkStartsWith(List<String> subArrays, List<String> superArrays){
+            for (String s1: subArrays){
+                boolean temp = false;
```

Comment:
`temp` isn't a very good variable name. Try to give every variable a descriptive name.